### PR TITLE
Adds User/Developer Documentation for BaSyx V2 

### PR DESCRIPTION
**Pull request by FriedJannik**
Co-authored-by: Aaron Zielstorff <aaron.zielstorff@iese.fraunhofer.de>

### CODE REVIEW

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/operation-delegation.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Operation Delegation
+An operation is a specific type of submodel element capable of performing server-side functions. However, in certain environments, this is not possible. Therefore, the operation can be delegated to an independent destination (such as an HTTP URL) to be executed. An example request body is given below.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/operation-delegation.md:2]: > An operation is a specific type of submodel element capable of (...)

&rarr; An operation is a specific type of submodel element, capable of (...)

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/operation-delegation.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+```
+
+To create an operation delegation, a Submodel that has an Operation(-SubmodelElement) must be present:
+-	To achieve the delegation functionality, a Qualifier of type "invocationDelegation" is added to the operation.
+-	As a value, the connection information (such as an HTTP URL) to the desired operation must be given.
+
+This means that operations can be delegated to endpoints of the same server as well as to external servers. For the front-end, it remains completely transparent whether an operation was called directly or delegated.
+
+
+The independent destination where the request is to be delegated should support the request with a parameter ([OperationVariable[]](https://github.com/eclipse-aas4j/aas4j/blob/2abf04bc01f80bceafa575cf85da429d5fe63918/model/src/main/java/org/eclipse/digitaltwin/aas4j/v3/model/OperationVariable.java#L31)) and provide the output in a strict format ([OperationVariable[]](https://github.com/eclipse-aas4j/aas4j/blob/2abf04bc01f80bceafa575cf85da429d5fe63918/model/src/main/java/org/eclipse/digitaltwin/aas4j/v3/model/OperationVariable.java#L31))
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/operation-delegation.md:24]: > (...) where the request is to be delegated should support the request with (...)

&rarr; (...) where the request is to be delegated **_to,_** should support ~the~ requests with (...)

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/operation-delegation.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+As of now, only delegation to HTTP URLs is supported.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/operation-delegation.md:26]: > As of now, only delegation to HTTP URLs is supported. 
&rarr; As of now, only delegation to HTTP POST endpoints is supported.

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/registry-integration.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+It also automatically removes the Descriptor from the Registry when the Submodel is removed from the Repository.
+
+To enable this feature, the following two properties should be configured:
+
+```
+basyx.submodelrepository.feature.registryintegration = {Submodel-Registry-Base-Url}
+basyx.externalurl = {Submodel-Repo-Base-Url}
+```
+
+This feature gets enable automatically when both of the above defined properties are configured, i.e., no external enabled/disabled property is required.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/features/registry-integration.md:12]: > This feature gets enable automatically when both of the above defined properties are configured, i.e., no external enabled/disabled property is required.

&rarr; > This feature **_is_** enabled ~automatically~ when both of the above  defined properties are **_set_**. I.e., no **_additional_** enabled/disabled property is required.

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+## Features
+- [Authorization](./features/authorization.md)
+- [MQTT](./features/mqtt.md)
+- [Operation Delegation](./features/operation-delegation.md)
+- [Registry Integration](./features/registry-integration.md)
+
+## Configuration
+### Server Configuration ![Required](https://img.shields.io/badge/required-true-red)
+This section configures the server port, application name and Submodel Repository Name for the Submodel Repository.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:16]: > This section configures the server port, (...)

&rarr; **_Configuration of_** the server port, (...)

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+```properties
+server.port=8081
+
+spring.application.name=Submodel Repository
+basyx.smrepo.name = sm-repo
+```
+---
+
+### Backend Configuration ![Required](https://img.shields.io/badge/required-true-red)
+Configure the backend storage. By default, it uses InMemory. Optionally, you can configure MongoDB.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:26]: > Configure the backend storage.

&rarr; **_Configuration of_** the backend storage.

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+spring.data.mongodb.database=submodels
+spring.data.mongodb.authentication-database=admin
+spring.data.mongodb.username=mongoAdmin
+spring.data.mongodb.password=mongoPassword
+```
+---
+
+
+### CORS Configuration ![Default](https://img.shields.io/badge/default-false-blue) ![Required](https://img.shields.io/badge/required-false-red)
+Configure CORS settings to specify allowed origins and methods.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:46]: > Configure CORS settings (...)

&rarr; **_Configuration of_** CORS settings (...)

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+```{warning}
+To use the component with the [BaSyx Web GUI](../../web_ui/index.md) this configuration is required.
+
+To do this, you need to add the URL of the Web GUI to the allowed origins.
+```
+
+```properties
+basyx.cors.allowed-origins=http://localhost:3000, http://localhost:4000
+basyx.cors.allowed-methods=GET,POST,PATCH,DELETE,PUT,OPTIONS,HEAD
+```
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:57]: please make clear that this is an example and the UI does not necessarily have the given endpoints

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+---
+
+### [Authorization](./features/authorization.md) ![Default](https://img.shields.io/badge/default-false-blue) ![Required](https://img.shields.io/badge/required-false-red)
+Enable and configure authorization features, such as Role-Based Access Control (RBAC) and JWT Bearer Token Provider.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:61]: > Enable and configure authorization (...)

&rarr; **_Configuring and enabling_** authorization (...)

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+basyx.feature.authorization.type = rbac
+basyx.feature.authorization.jwtBearerTokenProvider = keycloak
+basyx.feature.authorization.rbac.file = classpath:rbac_rules.json
+spring.security.oauth2.resourceserver.jwt.issuer-uri= http://localhost:9096/realms/BaSyx
+
+```
+---
+
+### [Registry Integration](./features/registry-integration.md) ![Required](https://img.shields.io/badge/default-false-blue) ![Default](https://img.shields.io/badge/required-false-red)
+These settings allow the configuration of registry integration and the external URL of the repository.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:73]: Why are these two introduced together? aren't those two different concerns?

File: docs/source/content/user_documentation/basyx_components/v2/aas_repository/index.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+spring.data.mongodb.port=27017
+spring.data.mongodb.database=aas
+spring.data.mongodb.authentication-database=admin
+spring.data.mongodb.username=mongoAdmin
+spring.data.mongodb.password=mongoPassword
+```
+---
+
+### [Registry Integration](./features/registry-integration.md)
+These settings allow the configuration of registry integration and the external URL of the repository.
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/aas_repository/index.md:45]: Why are these two introduced together? aren't those two different concerns?

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+## Swagger UI
+In the Swagger UI, you can find the API documentation for the Submodel Repository.
+
+You can also execute all the API calls directly from the Swagger UI.
+
+The Aggregated API endpoint documentation is available at:
+
+	http://{host}:{port}/v3/api-docs
+	
+The Aggregated Swagger UI for the endpoint is available at:
```
Comment (jannisjung) [inline docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:114]: > The Aggregated Swagger UI for the endpoint is available at:

&rarr; The Aggregated Swagger UI for the **_endpoints_** is available at:

**Review by jannisjung — CHANGES_REQUESTED** [review]
Here is a brief summary of the Review:

* The link to the component documentation in The [Development/BaSyx Java V2 Documentation](docs/source/content/user_documentation/developer/basyx_java_v2/index.md) should link to the OTS documentation but points to the developer documentation of the components.
* Many files of the OTS registry componet documentation read more like Developer documentation and should be recondisered to be moved or partially split so that only relevant Information for loading and configuring OTC components remains.
* I have also added some wording suggestions and notes on typos

File: docs/source/content/user_documentation/developer/basyx_java_v2/components.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+	http://{host}:{port}/actuator/health
+For a detailed documentation on its configuration possibilities, see [Spring Management Endpoint](management_endpoint.md) documentation.
+
+## CORS configuration
+Cross-Site Resource Sharing options can be configured under [application.properties](https://github.com/eclipse-basyx/basyx-java-server-sdk/blob/main/basyx.aasrepository/basyx.aasrepository.component/src/main/resources/application.properties#L26) with the following attribute:
+
+* Allowed Origin Patterns:<br>
+Comma-separated list of origin patterns to allow. Unlike allowed origins which only supports '*', origin patterns are more flexible (for example 'https://*.example.com') and can be used when credentials are allowed. When no allowed origin patterns or allowed origins are set, CORS support is disabled.
+  ```
+  basyx.cors.allowed-origins=http://localhost:8080, https://*.example.com
```
[the remark this answers is not in the document]
Comment (FriedJannik) [reply docs/source/content/user_documentation/developer/basyx_java_v2/components.md:26]: Those two properties (allowed-origin and allowed-origin-patterns) are combined in one, namely 'basyx.cors.allowed-origins'. So imo this should be fine

File: docs/source/content/user_documentation/basyx_components/v2/aas_registry/features/inmemory-storage.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+
+```xml
+
+	<dependency>
+		<groupId>org.eclipse.digitaltwin.basyx</groupId>
+		<artifactId>basyx.aasregistry</artifactId>
+	</dependency>
+```
+
+Then included, you can active it by either setting the active profile or the *registry.type* attribute:
```
[the remark this answers is not in the document]
Comment (FriedJannik) [reply docs/source/content/user_documentation/basyx_components/v2/aas_registry/features/inmemory-storage.md:13]: Profiles are shown below

File: docs/source/content/user_documentation/basyx_components/v2/aas_registry/features/inmemory-storage.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
```
[the remark this answers is not in the document]
Comment (FriedJannik) [reply docs/source/content/user_documentation/basyx_components/v2/aas_registry/features/inmemory-storage.md:14]: Has to be clarified with the team

File: docs/source/content/user_documentation/basyx_components/v2/aas_registry/features/paths.md [hunk 16]
```diff
@@ -0,0 +1,17 @@
+# Paths
```
[the remark this answers is not in the document]
Comment (FriedJannik) [reply docs/source/content/user_documentation/basyx_components/v2/aas_registry/features/paths.md:1]: Rework of Registry Documentation has to be done in a separate Task. We will need to discuss that in a meeting with all.

File: docs/source/content/user_documentation/basyx_components/v2/aas_registry/index.md [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+      - '8083:8080'
+    environment:
+      - SERVER_PORT=8080
+    volumes:
+      - ./basyx/aas-registry.yml:/workspace/config/application.yml
+    restart: always
+```
+
+```{important}
+The REST API and the client implementation will not be modified - if not a SNAPSHOT version - until a new major version is released or an update of the openAPI definition. All server-side classes and the plugins are not intended to be used as programming library. They could be updated or removed then a new minor version is released.
```
[the remark this answers is not in the document]
Comment (jannikf02) [reply docs/source/content/user_documentation/basyx_components/v2/aas_registry/index.md:101]: We're not sure either, we will delete it for now

File: docs/source/content/user_documentation/basyx_components/v2/aas_repository/index.md [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+- [Registry Integration](./features/registry-integration.md)
+- [MQTT](./features/mqtt.md)
+
+## Configuration
+
+### Server Configuration
+This section configures the server port, application name, error path and AAS Repo Name for the AAS Repository.
+```properties
+server.port = 8081
+server.error.path = /error
```
[the remark this answers is not in the document]
Comment (FriedJannik) [reply docs/source/content/user_documentation/basyx_components/v2/aas_repository/index.md:19]: This is only used here, we will remove it as it is Spring specific

File: docs/source/content/user_documentation/basyx_components/v2/cd_repository/index.md [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+```{warning}
+To use the component with the [BaSyx Web GUI](../../web_ui/index.md) this configuration is required.
+
+To do this, you need to add the URL of the Web GUI to the allowed origins.
+```
+
+```properties
+basyx.cors.allowed-origins = http://localhost:3000, http://localhost:4000
+basyx.cors.allowed-methods = GET,POST,PATCH,DELETE,PUT,OPTIONS,HEAD
+```
```
[the remark this answers is not in the document]
Comment (FriedJannik) [reply docs/source/content/user_documentation/basyx_components/v2/cd_repository/index.md:54]: Property does not exist, we adapt the allowed-origins to showcase this

File: docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md (hunk 10 above)
Comment (FriedJannik) [reply docs/source/content/user_documentation/basyx_components/v2/submodel_repository/index.md:73]: They belong together, you need this property solely for the Registry Integration

File: docs/source/content/user_documentation/developer/basyx_java_v2/components.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (jannisjung) [reply docs/source/content/user_documentation/developer/basyx_java_v2/components.md:29]: If this is the case, the documentation is contradictory as it explicitly mentions that `allowed-origins` is not as powerful as `allowed-origin-patterns`. At least that is how I understand it.

**Review by jannisjung — COMMENTED** [review]
Than you for your work. As far as I understand, the spring properties `allowed-origins` and `allowed-origin-patterns` are not interchangeable.

**Comment by FriedJannik** [comment]
> Than you for your work. As far as I understand, the spring properties `allowed-origins` and `allowed-origin-patterns` are not interchangeable.

Hi Jannis, regarding the spring management CORS properties you are right. (management....cors...allowed-origins[-patterns])
AFAIK in BaSyx we have our own implementation of CORS Properties,so they are united into one single Property

**Review by jannisjung — APPROVED** [review]
approved - thank you for your work and answers

**Pull request merged**